Shopify: Fill in recommended Product schema jsonld fields for SEO

masonmcelvain opened the pull request:
Injects the new jsonld we are building on the backend (see https://github.com/iFixit/ifixit/pull/40347) into the html `<head>`. We don't need to build the jsonld in JS anymore, so we can drop that code.

Also updates some testing data and a snapshot test.

## QA
Start the dev server and make sure the jsonld script tag gets added IFF the jsonld is not empty. Copy the `<head>` into [google's rich results tool](https://search.google.com/test/rich-results) and make sure there are no warnings or errors.

CC: @sterlinghirsh @JFleischhauer @GrantMatejka

Closes https://github.com/ifixit/ifixit/issues/39560

cdcline commented:
CR 👍

masonmcelvain commented:
The parent pull merged so in theory we should be able to test this now

GrantMatejka commented on src/lib/product_reviews/visualizer/index.js:
```diff
@@ -89,7 +109,16 @@ class Visualizer extends Component {
            "reviewCount": numReviews
          },
-         "review": [schemaReviews],
+         "brand": {
+            "@type": "Brand",
+            "name": "iFixit",
```
Will this be true for every product that uses this component?

GrantMatejka replied:
Yeah me neither, but if we don't have the data we should probably just not include it

masonmcelvain commented:
The parent pull failed on deploy so we can't QA this yet

zdmitchell commented:
CR 🎸 but deploy_block 🎸 on checking with someone about https://github.com/iFixit/product-page-components/pull/41#discussion_r733124854.

masonmcelvain committed:
Add fake jsonld data to example reviews json

We now get a jsonld prop from the backend that we can use in the ProductReviews component.
This is a copy of the jsonld that gets generated when you visit
https://www.cominor.com/Store/iPad/iPad-mini-LCD/IF122-000?o=1

masonmcelvain committed:
Inject jsonld prop into the html head

As of iFixit/ifixit#40347 we are creating this jsonld on the backend, so we don't need to build it in JS anymore.

If there is no jsonld available, the `jsonld` variable will be an empty object. In that case, we'll omit including the script tag entirely, because we have no jsonld to inject.

masonmcelvain commented:
QA ✌🏻 The jsonld gets injected into the < head> when you run the dev server, and copying the html into [Google's Rich Results Test tool](https://search.google.com/test/rich-results) passes with no warnings or errors.

- Ran `npm start`
- Inspected the < head> element and saw that only the nonempty jsonld was added to the head
![image](https://user-images.githubusercontent.com/52104630/139596054-417a81a3-2a07-45b9-96db-0baa687039d5.png)
- Copied the contents of the < head> into the rich results test, and confirmed the test passed
![image](https://user-images.githubusercontent.com/52104630/139596059-44c8ead8-fef7-4cbc-841f-39f25344be43.png)

Now that we've confirmed it looks like we want, I'm going to update the snapshot tests.

masonmcelvain committed:
test: Update product reviews snapshot test

We are passing an extra prop to the product reviews visualizer that can inject a script into the html head, so this snapshot needs updating.

masonmcelvain commented:
QA ✌🏻 carry forward, just updated a test

masonmcelvain commented:
un_dev_block ✌🏻 

un_deploy_block ✌🏻 We worked out where 'brand' data gets sourced from in iFixit/ifixit#40347

cdcline commented:
CR 👍

zdmitchell commented on src/lib/product_reviews/visualizer/index.js:
```diff
@@ -60,50 +60,16 @@ const ReviewCount = styled.h3`
-                  <script type="application/ld+json">{JSON.stringify(reviewJsonld)}</script>
-               }
-            </Helmet>
+            {Object.keys(jsonld).length && (
+               <Helmet>
+                  <script type="application/ld+json">{JSON.stringify(jsonld)}</script>
+               </Helmet>
+            )}
```
Doesn't React output 0 here if length is 0?

masonmcelvain replied:
You're totally right, thanks for catching that

masonmcelvain replied:
https://github.com/iFixit/product-page-components/pull/41/commits/b7454f3f0f26b25b06aa0f7bb14828dd75de189f

zdmitchell commented:
CR 🎸 but dev_block 🎸 on what I think might be a bug.

masonmcelvain committed:
fix: Use boolean literal to conditionally render

Returning a falsy expression causes the element after `&&` to be skipped, but returns the falsy expression.
This is why we were seeing `0` appear in the snapshot test.

Using a boolean should still skip the element in the falsy case, and won't output a 0.

https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

zdmitchell commented:
CR 🎸

cdcline commented:
CR 👍
